docs/review_checklist.md

```diff
@@ -30,7 +30,7 @@ Below is an example of the review checklist for the [Yellowbrick JOSS submission
 
 ### Documentation
 
-- **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
+- **A statement of need:** Do the authors clearly state what problems the software is designed to solve, who the target audience is, and its relation to prior works?
```

Comment:
@pdebuyl "relation to related work" seemed clumsy given the sentence structure, but I agree that "related work" is better language...

Reply:
Should we have this requirement (discussing related work) for both the project documentation and the JOSS paper, or would it be acceptable for it to only be in the JOSS paper?

My personal view is that it's necessary in the paper and nice to have in project documentation. However, it can be hard to maintain in project documentation because other packages are created and existing ones develop new features such that rationale can become stale. I think it's good for maintainers to stay informed about major capabilities of peers/competitors and end-to-end workflows, but perhaps it's too much to ask. I think it could be harmful for professional relationships if package X says something is a unique killer feature not available in package Y, and yet package Y does that thing in its next release but the page doesn't get updated for years. It could be resolved with a friendly email, but people don't always do that.

Reply:
In my opinion, it is ok to require it only in the article, as it is a snapshot of the software and of the landscape at the time of publication. Also, we don't have control on the future of the packages (and their doc) that we review.

Reply:
I'll agree with it being more important (and time bound) in the paper

Reply:
> and its relation to other work

I agree with @danielskatz that 'other' is the right thing here. Rationale we want people to mention other work (not by them) whereas 'prior' might suggest prior work by this team?

Reply:
Thanks, I've updated this PR. I don't know what to do with this sentence fragment appearing various places in `spec/fixture/`. I understand these are tests with previously published papers so perhaps the wording should not be updated because it's comparing with the state of the old paper. Or perhaps it's meant to be a dry-run with the content of the old paper in the new environment, but the CI action passed :shrug:.